### ISSUE TITLE
What's the right way to include inline JS from remote sources?

### ISSUE THREAD

**Issue by IonicaBizau** [substack/node-browserify#1658]
How can I include a remote script in the bundle? Similar to `inline` @import` from less.

**Comment by chinedufn** [comment]
You might 

- use [browserify-shim](https://github.com/thlorenz/browserify-shim#a-expose-global-variables-via-global) to expose your script. Without knowing what you're looking to do I'd guess that this could be the answer that you're looking for.

- create a [browserify transform](https://github.com/substack/browserify-handbook#transforms) to download your script during build time
  - Wouldn't recommend this unless it was necessary for some reason

**Comment by IonicaBizau** [comment]
@chinedufn The first link is quite useful!

Tho, what I need is to download the script if it's remote. So, by doing `b.add('https://example.com/foo.js');`, `foo.js` should appear in the bundle and be executed. It won't be required by any script.

**Comment by chinedufn** [comment]
@IonicaBizau ahh gotcha

So sorry if I'm misunderstanding, but are there any issues with downloading the script and then adding it to the bundle?

i.e.:

```
require('xhr').get('https://example.com/foo.js', function (err, resp) {
  if (err) { throw(err) }
  var myScript = resp.body
  b.add(myScript)
})
```

**Comment by IonicaBizau** [comment]
@chinedufn Oh, so, can we simply `add` a JavaScript code snippet? I guessed that works only for paths.

I'll try it.

**Comment by IonicaBizau** [comment]
Thanks! :cake:

**Comment by chinedufn** [comment]
Gah I think it might only work with paths as you said. @terinjokes do you know if there's a reason for this?

@IonicaBizau in the meantime maybe just `fs.writeFileSync` and then `b.add` that file?

**Comment by IonicaBizau** [comment]
@chinedufn Correct, but I was thinking there is a transform for that, something similar to this: https://github.com/dsdenes/browserify-cdnjs

While that looks good for `cdnjs`, I'd like to have one for any url. In my case, I don't really need to `require` the resources, but just to get them executed on the page.

**Comment by chinedufn** [comment]
@IonicaBizau ah cool, gotcha gotcha

I just [searched "browserify load script" on node-modules.com](http://node-modules.com/search?q=browserify+load+script) and came across [bromote](https://github.com/thlorenz/bromote)

Might that be of any use?

**Comment by MellowMelon** [comment]
Browserify allows you to feed in streams when you call methods like add. From front readme:
> If an entry file is a stream, its contents will be used. You should pass opts.basedir when using streaming files so that relative requires can be resolved.
